### DISCUSSION TITLE
Is it possible to check what mods installed on client

### DISCUSSION THREAD

**Discussion by semenishchev** [FabricMC/fabric-api#2055] [Mod Dev Support]
I need to check what mods installed on client that joins my server. Is it plugin messaging channel for handshake like forge has or so?

**Comment by warjort** [comment]
The network api is here:
https://github.com/FabricMC/fabric/tree/1.18.2/fabric-networking-api-v1/src/main/java/net/fabricmc/fabric/api
the wiki is here:
https://fabricmc.net/wiki/tutorial:networking

For custom login processing you probably want to start with this?
https://github.com/FabricMC/fabric/blob/1.18.2/fabric-networking-api-v1/src/main/java/net/fabricmc/fabric/api/networking/v1/ServerLoginConnectionEvents.java

I don't know what you are trying to achieve with the checking of client mods?
But inconsistencies between client/server are usually checked more concretely via registry synchronization
e.g. checking/ensuring blocks, entities, etc. agree on what ids to use or more "dynamically" sending recipes, world gen, etc. from server config when it changes

**Comment by Linguardium** [comment]
if this is to block the use of certain mods, be aware that it is trivial to change the mod id so checking the reported mods is useless
